src/main/java/frc/robot/commands/TeleopCommand.java

```diff
@@ -3,38 +3,52 @@
 import edu.wpi.first.wpilibj2.command.CommandBase;
 import frc.robot.io.Controls;
 import frc.robot.subsystems.drive.Drive;
+import frc.robot.subsystems.drive.Gyroscope;
 
 public class TeleopCommand extends CommandBase {
 
     private Drive drive;
+    private Gyroscope gyroscope;
     private Controls controls;
     private double DEADZONE = 0.11;
-    private boolean resetEncoders;
+    double desiredAngle;
+    boolean drivingStraight = false;
   
-    public TeleopCommand(Controls controls, Drive drive) {
+    public TeleopCommand(Controls controls, Drive drive, Gyroscope gyroscope) {
         this.controls = controls;
         this.drive = drive;
+        this.gyroscope = gyroscope;
+
+        addRequirements(controls);
+        addRequirements(drive);
+        addRequirements(gyroscope);
     }
 
     // Called just before this Command runs the first time
     @Override
     public void initialize() {
-        System.out.println("Initialized");
-
+        gyroscope.resetGyro();
     }
 
     // Called repeatedly when this Command is scheduled to run
     @Override
     public void execute() {
+        System.out.println(gyroscope.getGyroAngle());
         if (controls.getRightX(DEADZONE) == 0.0) {
```

Comment:
Having if statements to see what control button you're pressing defeats the purpose of command based programming. In command based programming we want to use the `whenPressed()` function in your IO class in order to let the system schedule the commands. Sorry that we didn't catch this when you merged your encoder code.

The appropriate way to handle this is check your turn modifier value in your arcade drive method and apply your drive straight code if it's equal to 0. It's the same idea except you're not checking your controls. This command should not be aware of the Controls class.

Reply:
This might be the reason your drive straight code doesn't work and makes it spin.